src/test/regress/expected/cpu_priority.out

```diff
@@ -0,0 +1,101 @@
+CREATE SCHEMA cpu_priority;
+SET search_path TO cpu_priority;
+SET citus.shard_count TO 4;
+SET citus.shard_replication_factor TO 1;
+SET citus.next_shard_id TO 11568900;
+-- This test depends on the fact that CI and dev machines don't have their
+-- resource limits configured in a way that allows raising CPU priority. This
+-- still tries to test as much functionality as possible by expecting certain
+-- error messages to appear, but it's not the environment production is
+-- supposed to be running as.
+CREATE USER cpu_priority_user;
+GRANT ALL ON SCHEMA cpu_priority to cpu_priority_user;
+CREATE TABLE t(a int);
+SELECT create_distributed_table('t', 'a');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+INSERT INTO t SELECT generate_series(1, 100);
+-- gives a warning because this is not allowed
+SET citus.cpu_priority = -20;
+WARNING:  could not set cpu priority to -20: Permission denied
+HINT:  Try changing the 'nice' resource limit by changing /etc/security/limits.conf for the postgres userand/or by setting LimitNICE in your the systemd service file (depending on how you start postgres).
+-- no-op should be allowed
+SET citus.cpu_priority = 0;
+-- lowering should be allowed
+SET citus.cpu_priority = 1;
+-- resetting should be allowed, but warn;
+RESET citus.cpu_priority;
+WARNING:  could not set cpu priority to 0: Permission denied
+HINT:  Try changing the 'nice' resource limit by changing /etc/security/limits.conf for the postgres userand/or by setting LimitNICE in your the systemd service file (depending on how you start postgres).
+SET citus.propagate_set_commands = local;
+BEGIN;
+    SET LOCAL citus.cpu_priority = 10;
+    SELECT count(*) FROM t;
+ count
+---------------------------------------------------------------------
+   100
+(1 row)
+
+    -- warning is expected here because raising isn't allowed by the OS
+COMMIT;
+WARNING:  could not set cpu priority to 0: Permission denied
```

Comment:
can we get into situations where we do not get these warnings? (do we need alternative output?)

Reply:
Only if you went through the effort of allowing niceness changes for processes. This can be trivially changed by running `ulimit -e 0`